capstone/src/test/java/com/google/step/servlets/RestaurantServletTest.java

```diff
@@ -0,0 +1,202 @@
+        ServletUnitClient sc = sr.newClient();
+
+        // This call to getResponse should throw an exception
+        WebRequest request = new GetMethodWebRequest("http://localhost:8080/restaurant?restaurantKey=0");
+        sc.getResponse(request);
+    }
+
+    /**
+     * Test the doPost method of RestaurantServlet by creating a request with random params
+     * then querying the mock Datastore to ensure the new restaurant was entered correctly.
+     * @throws IOException
+     * @throws SAXException if the ServletRunner parser cannot handle the request
+     */
+    @Test
+    public void testDoPost() throws IOException, SAXException {
+        DatastoreService datastoreService = DatastoreServiceFactory.getDatastoreService();
+        String name =  "Wildfire";
+        String story = "Swanky American chain serving steak, chops & seafood, plus burgers, sides & cocktails.";
+        String cuisineList = "Steakhouse, American";
+        String phone = "8472797900";
+        String website = "https://www.wildfirerestaurant.com";
+
+        // User must be logged in to post a restaurant
+        helper.setEnvIsAdmin(false).setEnvIsLoggedIn(true);
+        helper.setEnvEmail("wildfire@gmail.com");
+        helper.setEnvAuthDomain("google");
+
+        // Create a client to run the servlet
+        ServletRunner sr = new ServletRunner();
+        sr.registerServlet("restaurant", RestaurantServlet.class.getName());
+        ServletUnitClient sc = sr.newClient();
+
+        // Set params for the post request
+        WebRequest request = new PostMethodWebRequest("http://localhost:8080/restaurant");
+        request.setParameter("name", name);
+        request.setParameter("cuisine", cuisineList);
+        request.setParameter("story", story);
+        request.setParameter("phone", phone);
+        request.setParameter("website", website);
+        sc.getResponse(request);
+
+        // Simple query of the mock Datastore to check if the restaurant was entered correctly
+        Query query = new Query("RestaurantInfo")
+                .setFilter(new Query.FilterPredicate(
+                        "name", Query.FilterOperator.EQUAL, name));
+        PreparedQuery results = datastoreService.prepare(query);
+        Entity resultEntity = results.asSingleEntity();
+
+        assertEquals(name, resultEntity.getProperty("name"));
+        assertEquals("[Steakhouse,  American]", resultEntity.getProperty("cuisine").toString());
+        assertEquals(story, resultEntity.getProperty("story"));
+        assertEquals(phone, resultEntity.getProperty("phone"));
+        assertEquals(website, resultEntity.getProperty("website"));
+    }
+
+    /**
+     * Test the doPost method of RestaurantServlet with a user who is not logged in
+     * to ensure it will throw an exception.
+     * @throws IOException
+     * @throws SAXException if the ServletRunner parser cannot handle the request
+     */
+    @Test(expected = HttpException.class)
```

Comment:
Should this be checking for a specific http code?

Reply:
Thanks Megan, I changed it to check for the specific SC_FORBIDDEN code.